### DISCUSSION TITLE
Running QEMU/KVM VMs in K8S - a new development

### DISCUSSION THREAD

**Discussion by shimpa1** [akash-network/community#534] [Ideas]
At KubeCon we had the privilege to talk to an engineer from Kubevirt - a project developing a framework that would allow to run KVM/QEMU VMs inside K8S.
@cloud-j-luna actually found them and we had a chat. We specifically asked them whether root privileges are required for VMs and the answer was no.
To explain further, there is a management pod that runs in the cluster that does have root privileges and it spawns QEMU VMs that do not need root privileges.
https://github.com/kubevirt/kubevirt
Might be worth to explore this option.

cheers,

Shimpa

**Comment by cloud-j-luna** [comment]
There is already a ticket for this feature from last year. https://github.com/akash-network/support/issues/172 👍🏻

**Comment by shimpa1** [comment]
Great! No comments on that?
I actually believe this is an excellent feature.

**Reply by ghost** [reply]
Loving this idea! I know it was possible back then.

**Reply by cloud-j-luna** [reply]
> Great! No comments on that?
> 
> I actually believe this is an excellent feature.

It is indeed a great feature. I just want to raise some concerns with this technology. I'm not sure how many providers are running on virtualized environments, but those would not be fit to run Kubevirt as it adds another virtualizarion layer on top of the existing one, meaning it is best fit for bare metal environments. At the very least tenants should be aware of this as they have different levels of performance.

Nonetheless a discovery should be done to further spec this possible feature and its viability as a new resource on the network.

**Comment by faddat** [comment]
we should do this :)

**Comment by Zblocker64** [comment]
@shimpa1 @cloud-j-luna @andy108369 has there been any more thought put into this? I would love to see this come to fruition.

**Comment by mmchougule** [comment]
Hey @shimpa1 @faddat @cloud-j-luna , let's work on this integration together. I would love to enable MicroVMs to be deployed on Akash using Kata containers. We are building a Hypervisor Abstraction Interface that can be used to build on top of Akash. Are most Akash providers already running inside virtualized environments or can we find some bare metal ones with KVM support?

**Comment by 31trainman** [comment]
Let's take this to the akash discord and see what we can do together
